### PR TITLE
Port location filter to mobx

### PR DESCRIPTION
**Pull request by na9da**
Test config:

```
{
  "catalog": [
    {
      "name": "Sentinel - moment lines",
      "type": "wms",
      "layers": "s2_ard_granule_nbar_t",
      "url": "https://ows.dea.ga.gov.au/",
      "chartType": "momentPoints",
      "opacity": 1,
      "timeFilterPropertyName": "data_available_for_dates"
    }
  ]
}
```

### CODE REVIEW

File: lib/Traits/TimeFilterTraits.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import LatLonHeightTraits from "./LatLonHeightTraits";
+import objectTrait from "./objectTrait";
+import primitiveTrait from "./primitiveTrait";
+import mixTraits from "./mixTraits";
+
+export class TimeFilterCoordinates extends mixTraits(LatLonHeightTraits) {
+  @primitiveTrait({
+    type: "number",
+    name: "x",
+    description: "X coordinate of the tile"
```
Comment (rowanwins) [inline lib/Traits/TimeFilterTraits.ts:11]: These traits are very specific to how WMS works - I wonder if we should name them accordingly?

Comment (rowanwins) [reply lib/Traits/TimeFilterTraits.ts:11]: Or perhaps make reference to WMS in the description. 

The focus should be on the `LatLonHeightTraits` as that's what we're filtering by, but here it's pretty easy to miss that and get confused by the `x`, `y`, `level` which are really WMS specific.

**Review by rowanwins — APPROVED** [review]
This all works as it should, good job @na9da 

My only comment is that some of the trait naming or descriptions could perhaps be tweaked as currently it sounds more generic than it is in reality.  There are some really WMS specific stuff in there.

File: lib/Traits/TimeFilterTraits.ts (hunk 1 above)
Comment (na9da) [reply lib/Traits/TimeFilterTraits.ts:11]: Have renamed the traits and added a pointer to the cesium doc in the trait description.

Comment (rowanwins) [reply lib/Traits/TimeFilterTraits.ts:11]: As per discussion with Nanda it turns out we're just following the approach of Cesium for pickFeatures

**Pull request merged**